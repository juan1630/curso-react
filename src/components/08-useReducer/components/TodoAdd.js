import React from 'react';
import { useForm } from '../../../hooks/useHook'

export const TodoAdd = ({handleAddTodo}) => {

    const [ { description}, handleInputChange, reset  ] =  useForm({
        
        description:''
   
       });


       const handleOnSubmit = (e) =>{ 
            
        e.preventDefault();
        if( description.trim().length < 1 ){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }


        handleAddTodo( newTodo)
        // reseteamos el form 
        reset();

       
    }
   
    return (
        <>
             <h3 className="agregar" >
                        Agregar todo
                   </h3>


                   <hr/>

                    <form onSubmit={  handleOnSubmit }  >


                        <input 
                        type="text"
                        className="form-control"
                        name="description"
                        placeholder="Comer..."
                        autoComplete="off"
                        onChange={ handleInputChange }
                        value={ description }
                        />


                        <button className="btn btn-success btn-block" > Agregar </button>

                    </form>
        </>
    )
}
