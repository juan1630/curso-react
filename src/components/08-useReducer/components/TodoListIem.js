import React from 'react'


export const TodoListIem = ({ todo, i, handleDelete, handleToggle }) => {
    // console.log(todo, i, handleDelete, handleToggle )
    return (
        <div>
             <li  onClick={ ()=> { handleToggle(todo.id) } } className={ `${todo.done && 'complete'} ` }  key={todo.id} >  { i +1 } - {todo.desc} 
                            <hr/>
                            <button 
                            
                            className="btn btn-danger"
                            onClick={ () => { handleDelete( todo.id ) } }
                            > 

                                Borrar
                            </button>
                        
                          </li>
        </div>
    )
}
