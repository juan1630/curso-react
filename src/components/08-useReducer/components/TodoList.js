import React from 'react';
import { TodoListIem } from './TodoListIem'

export const TodoList = ( { todos = [] , handleDelete, handleToggle} ) => {

    console.log( todos );
    
    return (

               <ul className="todos-items" >
                {
                    todos.map( (todo, i) => (
                        
                        <TodoListIem 
                        key={ todo.id }
                        todo={ todo} 
                         i={ i }   
                        handleDelete ={handleDelete} 
                        handleToggle={handleToggle} />
                       
                    ))
                }
            </ul>
    )
}
