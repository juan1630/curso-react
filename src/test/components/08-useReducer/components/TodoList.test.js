const { shallow } = require("enzyme");
import { demoTodos } from '../../../fixture/demoTodos'

import {TodoList}  from '../../../../components/08-useReducer/components/TodoList'

describe('Pruebas en el todoList', ()=> {
    
    // definimo como funciones
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow( <TodoList
                                todos={ demoTodos }
                                handleDelete
                                handleToggle
                            />);



    test('Debe de mostrar correctamente el componente', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('Debe de tener 2 TodoListItem', ()=> {
        // no dejar espacios en blanco
        expect( wrapper.find('TodoListIem').length ).toBe(demoTodos.length )
    });


    test('Debe de tener 2 TodoListItem', ()=> {
        // no dejar espacios en blanco
        expect( wrapper.find('TodoListIem').length ).toBe(demoTodos.length )
        
        expect( wrapper.find('TodoListIem').at(0).prop('handleDelete')  ).toBe(true);

    });

})