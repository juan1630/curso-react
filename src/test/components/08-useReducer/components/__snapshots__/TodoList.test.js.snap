// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el todoList Debe de mostrar correctamente el componente 1`] = `
<ul
  className="todos-items"
>
  <TodoListIem
    handleDelete={true}
    handleToggle={true}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListIem
    handleDelete={true}
    handleToggle={true}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
